File: 1528-kids-with-the-greatest-number-of-candies/solution.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& v, int extraCandies) {
        
        auto mx = max_element(v.begin(),v.end());
        vector<bool> res ;
        for(auto &i : v){
            if(i+extraCandies >= *mx) res.push_back(true);
            else res.push_back(false);
        }
        return res ;
    }
};

File: 1528-kids-with-the-greatest-number-of-candies/solution.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        
        int mx = 0 ;
        int n = candies.length ;
        List<Boolean> res = new ArrayList<>();

        for(int i = 0 ; i  < n ; i++){
            mx = Math.max(mx,candies[i]);
        }

        for(int i = 0 ; i < n ; i++){

            if(extraCandies + candies[i] >= mx)
                res.add(true);
            else res.add(false);    
            
        }

        return res ;
    }
}
